packages/cli/commons/src/npm/npf.ts

```diff
@@ -45,7 +45,7 @@ function isPackageInstalled(packageSpec: string): boolean {
   let packageJsonFromRequire = getPackageJsonFromLazyLoadDirectory(packageName);
   return packageVersion
     ? satisfies(packageJsonFromRequire?.version ?? '', packageVersion)
-    : true;
+    : Boolean(packageJsonFromRequire);
```

Comment:
- [ ] Add test case: "what if the package is not installed, and the required spec does not have a version"